### PR TITLE
Simplify FE_BDM::get_dpo_vector() somewhat.

### PR DESCRIPTION
**Pull request by bangerth**
In particular, make it a bit easier to read.

### CODE REVIEW

File: source/fe/fe_bdm.cc [hunk 1]
```diff
@@ -231,21 +231,21 @@ template <int dim>
...
-  if (deg>1)
-    interior_dofs = dim * PolynomialSpace<dim>::compute_n_pols(deg-1);
 
-  std::vector<unsigned int> dpo(dim+1);
-  dpo[dim-1] = dofs_per_face;
-  dpo[dim]   = interior_dofs;
+  std::vector<unsigned int> dpo(dim+1, 0u);
+  dpo[dim]   = (deg > 0 ?
+                dim * PolynomialSpace<dim>::compute_n_pols(deg-1) :
+                0u);
```
Comment (kronbichler) [inline source/fe/fe_bdm.cc]: Is `PolynomialSpace<dim>::compute_n_pols(0)` always `0`? You replaced a check `deg>1` by `deg>0`, so to be equivalent, this must be the case...

Comment (bangerth) [inline source/fe/fe_bdm.cc]: Yes, that is clearly wrong :-( Not sure what I was smoking. Thanks for catching it.

Comment (bangerth) [inline source/fe/fe_bdm.cc]: (Well, I do live in Colorado now where certain things are legal these days. But I can assure that nothing of that sort was involved.)

**Comment by bangerth** [comment]
Corrected patch posted.

**Pull request merged**